src/basicfuns.jl

```diff
@@ -153,8 +153,37 @@ Return `log(1+exp(x))` evaluated carefully for largish `x`.
 This is also called the ["softplus"](https://en.wikipedia.org/wiki/Rectifier_(neural_networks))
 transformation, being a smooth approximation to `max(0,x)`. Its inverse is [`logexpm1`](@ref).
 """
-log1pexp(x::Real) = x < 18.0 ? log1p(exp(x)) : x < 33.3 ? x + exp(-x) : oftype(exp(-x), x)
-log1pexp(x::Float32) = x < 9.0f0 ? log1p(exp(x)) : x < 16.0f0 ? x + exp(-x) : oftype(exp(-x), x)
+function log1pexp(x::Real)
+    t = log1p(exp(-abs(x)))
+    return x ≤ 0 ? t : t + x
+end
+
+function log1pexp(x::Union{Float16,Float32,Float64})
+    #= Faster implementation for floats. Based on Mächler (2012), available at:
+    https://cran.r-project.org/web/packages/Rmpfr/vignettes/log1mexp-note.pdf =#
```

Comment:
Can you move this reference to the docstring instead, similar to `log1mexp`?

Reply:
Yep